cpp/src/cuML.hpp

```diff
@@ -33,16 +33,21 @@ class cumlHandle_impl;
  * @brief Handle to manage resources needed by cuML algorithms.
  */
 class cumlHandle {
+ private:
+  //TODO: What is the right number?
+  static constexpr int _default_num_internal_streams = 3;
```

Comment:
I would just keep this one. am not sure how the number 3 was conjured in first place. Do we have a reason for '3' ? 
By default everything has to be in the cuml_handle stream but if more than one stream is needed, user can ask for that either in constructor or at a later stage in algorithm.

Reply:
Good catch. This number was just borrowed from the initial version that Jiri had introduced. :)

I'd probably say even better to change this to 0. So, for any *extra* streams required they need to be explicitly queried. What say?

@cjnolet @dantegd and @JohnZed any objections?

Reply:
yes exactly. I agree. extra streams need to be explicit.

Reply:
0 seems to be a reasonable default, since this is enabling a special case where algorithms can take advantage of multiple stream internally.

Reply:
Done.